sdk/armada-protocol-common/src/common/interfaces/IArmadaPool.ts

```diff
@@ -0,0 +1,44 @@
+import { IPool, PoolDataSchema, PoolType } from '@summerfi/sdk-common/common'
+import { z } from 'zod'
+import { IArmadaPoolId, isArmadaPoolId } from './IArmadaPoolId'
+
+/**
+ * @interface IArmadaPool
+ * @description Interface for an ID of an Armada Protocol pool (fleet)
+ */
+export interface IArmadaPool extends IPool, IArmadaPoolData {
+  /** Signature used to differentiate it from similar interfaces */
+  readonly _signature_1: 'IArmadaPool'
+  /** ID of the pool */
+  readonly id: IArmadaPoolId
+
+  // Re-declaring the properties with the correct types
+  readonly type: PoolType
```

Comment:
why redeclaring since it's the same as the base class?

Reply:
shouldn't this be PoolType.Armada since it's set in the class implementation constructor and only valid PoolType for armada pool

Reply:
Discuss offline, I'll try to refactor this in a next PR. I've added a task to not forget